### PR TITLE
Closes #7162 - Feature - Orchard.DeveloperTools - Add toggle from front end

### LINKED ISSUE #7162: Feature - Orchard.DeveloperTools - Add toggle from front end
When I'm developing a site sometimes I will want the shape tracer on and sometimes I will want to work on the css / html with a clean view.

Adding a button to activate and deactivate the shape tracer from its front end ui makes this process a lot quicker.

The idea behind the feature is to allow quick toggling while developing, you would still disable the module at the end of development as you normally would.

## New settings tab
In the editor there is a new tab which lets you deactivate the shape tracer:

image

The page reloads and the page no longer has all the wrappers cluttering up the html markup.

You can quickly re-activate it when you need it again by clicking the activate button in the front end:

image

## Admin panel settings
You can also edit the setting in the new Shape Tracing section of the settings panel:

image

image

## Visual amendments
The toolbar along the top "toolbar" has been given additional height for modern displays:

image

A notifier has been added which lets you see at-a-glance if the system is currently active:

image

image

The menu toggle button has been remade in CSS, better supporting modern high pixel density displays:

image image

## Enhanced click zone

The entire top bar is now a click zone to toggle the panel up and down:

image

## Automatic re-enable

Each time the module itself is enabled the active setting is also reset so that it turns on:

image

Less confusion for users who disabled the module while the feature was deactivated as well.

## Further feature ideas
These are the remaining items on the todo list for future enhancements, I welcome any feedback:

  - Review - ShapeTracingFactory.cs - Created() - Is it expensive / being run still when deactivated?
  - Review security around allowing shape tracing enable from UI
  - Add layer rule / conditions support
  - Add option to enable shape tracer to be used against the admin panel theme
  - Apply assets.json gulp pipeline support
  - Does it need import / export now?

### PR DESCRIPTION
**Pull request by rtpHarry**
Closes #7162 - Feature - Orchard.DeveloperTools - Add toggle from front end

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.DesignerTools/Events/FeatureEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        public void Disabling(Feature feature) {
+
+        }
+
+        public void Enabled(Feature feature) {
+            if(feature.Descriptor.Id != "Orchard.DesignerTools") {
+                return;
+            }
+
+            // reset it to active each time the module is turned back on - less confusing for users
```
Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.DesignerTools/Events/FeatureEventHandler.cs]: Please use proper sentences for comments, starting with a Capital word and ending with a full stop.

File: src/Orchard.Web/Modules/Orchard.DesignerTools/Handlers/ShapeTracingSiteSettingsPartHandler.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            Filters.Add(new TemplateFilterForPart<ShapeTracingSiteSettingsPart>("ShapeTracingSiteSettings", "Parts/ShapeTracing.ShapeTracingSiteSettings", "ShapeTracing"));
+
+            OnInitializing<ShapeTracingSiteSettingsPart>(AssignDefaultValues);
+        }
+
+        private void AssignDefaultValues(InitializingContentContext context, ShapeTracingSiteSettingsPart part) {
+            part.IsShapeTracingActive = true;
+        }
+
+        public Localizer T { get; set; }
```
Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.DesignerTools/Handlers/ShapeTracingSiteSettingsPartHandler.cs]: Please move public properties before public methods.

File: src/Orchard.Web/Modules/Orchard.DesignerTools/Handlers/ShapeTracingSiteSettingsPartHandler.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        protected override void GetItemMetadata(GetContentItemMetadataContext context) {
+            if (context.ContentItem.ContentType != "Site")
+                return;
+
+            base.GetItemMetadata(context);
+
+            var groupInfo = new GroupInfo(T("Shape Tracing"));
+            groupInfo.Id = "ShapeTracing";
```
Comment (rtpHarry) [inline src/Orchard.Web/Modules/Orchard.DesignerTools/Handlers/ShapeTracingSiteSettingsPartHandler.cs]: I'm trying to get it changed in #7155 so people don't have to use it.

**Comment by rtpHarry** [comment]
@sfmskywalker thanks for the feedback, i've pushed some amends

**Pull request merged**